### ISSUE TITLE
launch.json "type" field required to be outside of OS specific section

### ISSUE THREAD

**Issue by sholmes74** [Microsoft/vscode#70508]
Issue Type: <b>Bug</b>

I'm trying to develop a cross-platform C++ app, but I have to duplicate every configuration for Mac and Windows to switch the "type" field between "cppdbg" for Mac and "cppvsdbg" for Windows.  Putting the "type" field within the "osx" or "windows" section doesn't work and won't launch the app.  This largely defeats the purpose of even having an OS specific section if it's only valid for certain fields and one of those isn't the "type" of debugger.  

This section below should be valid, every field here works except for the "type" field (althought the syntax checker complains about missing the "program" field, it's not breaking problem):
{
     "name": "Cross-platform App",
     "request": "launch",
     "osx": {
          "type": "cppdbg",
          "program": "${workspaceFolder}/Mac/myapp.app/Contents/MacOS/myapp",
          "cwd": "${workspaceFolder}/Mac/myapp.app/Contents/MacOS",
          "MIMode": "lldb"
     },
     "windows": {
         "type": "cppvsdbg",
         "program": "${workspaceFolder}\Windows\myapp.exe",
         "cwd": "${workspaceFolder}\Windows"
     },
     "args": [
         "--arg1"
     ],
     "stopAtEntry": false,
     "environment": [],
     "externalConsole": true
},

This configuration below should also be valid and should only show in the drop down on Mac and not Windows:
{
     "osx": {
         "name": "OSX-Only App",
         "request": "launch",
         "type": "cppdbg",
         "program": "${workspaceFolder}/Mac/myapp.app/Contents/MacOS/myapp",
         "cwd": "${workspaceFolder}/Mac/myapp.app/Contents/MacOS",
         "MIMode": "lldb"
         "args": [
             "--arg1"
         ],
         "stopAtEntry": false,
         "environment": [],
         "externalConsole": true
    }
} 

VS Code version: Code 1.32.2 (e64cb27b1a0cbbc3f643c9fc6c7d93d6c6509951, 2019-03-13T01:56:08.693Z)
OS version: Darwin x64 18.2.0

**Comment by roblourens** [comment]
The rest of the schema depends on the value of `type`, so it's a special case...

**Comment by isidorn** [comment]
Yes, 'type' is a special case which we probably will not allow to be os-specific. Though leaving this open as a feature request for now.
